### PR TITLE
Copy key in a way which preserves metadata

### PR DESCRIPTION
**Pull request by adamtheturtle**
See FLOC-1471 comments for information and linked Buildbot PR.

### CODE REVIEW

File: admin/aws.py [hunk 1]
```diff
@@ -126,11 +126,23 @@ def perform_copy_s3_keys(dispatcher, intent):
     """
     s3 = boto.connect_s3()
     destination_bucket = s3.get_bucket(intent.destination_bucket)
+    source_bucket = s3.get_bucket(intent.source_bucket)
     for key in intent.keys:
-        destination_bucket.copy_key(
-            new_key_name=intent.destination_prefix + key,
-            src_bucket_name=intent.source_bucket,
-            src_key_name=intent.source_prefix + key)
+        if '/_images/' in key and key.endswith('.svg'):
```
Comment (itamarst) [inline admin/aws.py]: Check for `/_images/` seems fragile? And I can't imagine we'd ever upload something to docs that ends with `.svg` that isn't a SVG.

Comment (adamtheturtle) [inline admin/aws.py]: I didn't want to touch fonts as I'd heard no complaints about those, which is why I had this here.

File: admin/aws.py [hunk 2]
```diff
@@ -126,11 +126,23 @@ def perform_copy_s3_keys(dispatcher, intent):
...
+            # s3cmd isn't great at choosing Content-Type. This fixes the one
+            # symptom of that we know of.
```
Comment (tomprince) [inline admin/aws.py]: I've done some poking around, and it appears that with the latest version of `s3cmd`, if we *don't* pass `--no-mime-magic`, it does the right thing for svg.

**Comment by tomprince** [comment]
I think, given that `s3cmd` can be made to do the right thing, this branch is unneeded.

**Comment by tomprince** [comment]
I wonder if, rather than checking just some files, and fixing their mime time, if it would make sense to just override the mime-type and not try and depend on something external to get the rest right.

The following is a list of the extant extensions:
- eot
- gif
- html
- jpg
- js
- css
- png
- sh
- svg
- ttf
- txt
- woff
- yml

**Comment by itamarst** [comment]
Sounds good.

**Pull request merged**